packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -331,6 +334,12 @@ export abstract class FluidDataStoreContext extends TypedEventEmitter<IFluidData
         }
     }
 
+    public abstract delete(): void;
```

Comment:
You can set this.deleted = true in this function itself. The local data store can then override this and call super.delete() after sending the log.

Reply:
Sure I can do that

Reply:
Are you liking your own comment to keep track of whether you took care of them? :D :D

Reply:
Yes